### PR TITLE
feat: add database header in storage request

### PR DESCRIPTION
**Pull request by mrrtree**
- Add `RequestContext` in all strorage service request, and `RequestContext` required `database`

### CODE REVIEW

**Review by ShiKaiWi — COMMENTED** [review]
LGTM

File: protos/storage.proto [hunk 1]
```diff
@@ -17,8 +17,13 @@ service StorageService {
   rpc PromQuery(prometheus.PrometheusQueryRequest) returns (prometheus.PrometheusQueryResponse) {}
 }
 
+message RequestHeader {
```
Comment (jiacai2050) [inline protos/storage.proto:20]: RequestContext

IMO A request doesn't have header and body part

Comment (mrrtree) [reply protos/storage.proto]: done

**Review by jiacai2050 — APPROVED** [review]
LGTM

**Pull request merged**